AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/Subscribe/DataStoreObserveQueryOperation.swift

```diff
@@ -211,6 +211,13 @@ class ObserveQueryTaskRunner<M: Model>: InternalTaskRunner, InternalTaskAsyncThr
             })
     }
 
+    /// If the `storageEngine` does sync from remote, the initial batch should
+    /// collect snapshots based on time / snapshots received.
+    /// If it doesn't, it should publish each snapshot without waiting.
+    private var shouldBatchDuringSync: Bool {
+        (storageEngine as? StorageEngine)?.syncsFromRemote ?? true
```

Comment:
would it make sense to add the computed property `syncsFromRemote` to `StorageEngineBehavior` (potentially a different name like `syncEnabled`)? Right now, using a mock would fail the `storageEngine as? StorageEngine` check, right?

Reply:
Unless the mock was a `StorageEngine` subclass - which isn't possible as `StorageEngine` is marked `final` - it would fail to cast in that case.

Adding it as a requirement to `StorageEngineBehavior` makes sense.

Reply:
Addressed in [dd0e0be](https://github.com/aws-amplify/amplify-swift/pull/3214/commits/dd0e0bed6a61032aefa913abf7a05d236b7acf3f)